backend/src/test/java/com/engineering/thesis/backend/serviceIntegrationTests/AppointmentCrudTests.java

```diff
@@ -0,0 +1,123 @@
+package com.engineering.thesis.backend.serviceIntegrationTests;
+
+import com.engineering.thesis.backend.exception.CreateObjException;
+import com.engineering.thesis.backend.model.Appointment;
+import com.engineering.thesis.backend.model.Doctor;
+import com.engineering.thesis.backend.model.Patient;
+import com.engineering.thesis.backend.model.User;
+import com.engineering.thesis.backend.repository.AppointmentRepository;
+import com.engineering.thesis.backend.serviceImpl.AppointmentServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class AppointmentCrudTests {
+
+    @Mock(lenient = true)
+    private AppointmentRepository appointmentService;
```

Comment:
Naming it's strange but it's making reading inside methods much easier.

Reply:
And even move all model dummy objects to seperate class as resource like I did with roles.

Reply:
For me giving name service for repository is bad practise. When I'm reading code and see service I expect methods from service classes, but I see repository method - misleading. Improve that on correct name.